src/libprojectM/Renderer/MilkdropNoise.cpp

```diff
@@ -108,23 +197,26 @@ void MilkdropNoise::generate2D(int size, int zoomFactor, uint32_t** textureData)
                 }
             }
         }
-
     }
+
+    return textureData;
```

Comment:
This is performing an additional (unnecessary) copy of textureData. I suggest either allocating textureData into a smart pointer and returning that or passing in the std::vector by reference instead to avoid the copy.

Reply:
I'm well over 99% sure most modern compilers will use always copy elision here, as it's a rather simple case. So using smart pointers or a reference would increase code complexity, but not give any advantages.

At least on my machines, I've checked this and `.data()` returns the same memory address inside the function (fater `resize()` and in the calling function as well in both release and debug builds, so no data has been copied to another memory location.

Reply:
I'm glad the compiler optimizes this away, but it's still allocating a vector on the stack, then returning that vector through a function call, which according to the spec will call the copy constructor. If you don't want to fix it that's fine (this is done during preset initialization so it doesn't really affect runtime performance) but it's not best practice.

Reply:
If you'd check this, you'll see that the compiler turns it into a move operation, and not even that in the `std::move` sense - the compiler will simply make the variable it's being assigned to point to the stack object. Even if the vector itself would be copied, it's just a pointer and a size, so 16 bytes to copy on 64 bit systems, which is actually exactly the size of `std::vector<>`. Not that much of a hassle.